comp/dogstatsd/server/rc.go

```diff
@@ -0,0 +1,105 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package server implements a component to run the dogstatsd server
+package server
+
+import (
+	"encoding/json"
+	"slices"
+
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+)
+
+type statsdBlocklistUpdate struct {
+	BlockedMetrics blockedMetrics `json:"blocked_metrics"`
+}
+
+type blockedMetrics struct {
+	ByName byName `json:"by_name"`
+}
+
+type byName struct {
+	ConfigurationID int           `json:"configuration_id"`
+	Metrics         []metricEntry `json:"values"`
+}
+
+type metricEntry struct {
+	Name string `json:"metric_name"`
+}
+
+func (s *server) onBlocklistUpdateCallback(updates map[string]state.RawConfig, applyStateCallback func(string, state.ApplyStatus)) {
+	s.log.Debug("onBlocklistUpdateCallback received updates:", len(updates))
+
+	if len(updates) == 0 {
+		return
+	}
+
+	var blocklistUpdates []blockedMetrics
+
+	// unmarshal all the configurations received from
+	// the RC platform
+	// ------------------------
+
+	if len(updates) > 0 {
+		// unmarshal all configs that can be unmarshalled
+		for configPath, v := range updates {
+			s.log.Info("received:", string(v.Config))
+			var config statsdBlocklistUpdate
+			if err := json.Unmarshal(v.Config, &config); err != nil {
+				applyStateCallback(configPath, state.ApplyStatus{
+					State: state.ApplyStateError,
+					Error: "error unmarshalling payload",
+				})
+				s.log.Error("can't unmarshal received config:", err)
+				continue
+			}
+			if len(config.BlockedMetrics.ByName.Metrics) == 0 {
+				s.log.Warn("received a configuration with no metric")
+				continue
+			}
+			blocklistUpdates = append(blocklistUpdates, config.BlockedMetrics)
+		}
+	}
+
+	// sort by the configuration ID
+	// ------------------------
+
+	slices.SortFunc(blocklistUpdates, func(a, b blockedMetrics) int {
+		return a.ByName.ConfigurationID - b.ByName.ConfigurationID
+	})
```

Comment:
What is the reason for sorting here? The next block combines everything into a single set, the order is not important for that.

Reply:
The whole reason is to have a stable behaviour with regards to the processing order, especially in the future since the schema and content of these configurations might change. Do you think I'm being over-zealous? I agree it's not necessary for now.

Reply:
I think to Vickenty's point: we already union the metrics from each update by virtue of how we write them into the map `m`, so their order here doesn't matter.